modules/cli/src/main/scala/scala/cli/commands/Update.scala

```diff
@@ -50,48 +52,46 @@ object Update extends ScalaCommand[UpdateOptions] {
     }
   }
 
-  def update(options: UpdateOptions, scalaCliBinPath: os.Path) = {
-    val currentVersion = {
-      val res = os.proc(scalaCliBinPath, "version").call(cwd = os.pwd, check = false)
-      if (res.exitCode == 0)
-        res.out.text().trim
-      else
-        "0.0.0"
+  lazy val updateInstructions: String =
+    s"""Your Scala CLI version is outdated. The newest version is $newestScalaCliVersion
+       |It is recommended that you update Scala CLI through the same tool or method you used for its initial installation for avoiding the creation of outdated duplicates.""".stripMargin
+
+  def update(options: UpdateOptions, maybeScalaCliBinPath: Option[os.Path]): Unit = {
+    val maybeCurrentVersion = maybeScalaCliBinPath.map {
+      scalaCliBinPath =>
+        val res = os.proc(scalaCliBinPath, "version").call(cwd = os.pwd, check = false)
```

Comment:
It's seems, that you changed `Version` command and now prints 
```
s"$appName version ${Version.version}" + Version.detailedVersionOpt.fold("")(
       " detailed version: (" + _ + ")"
     )
```
so this logic won't be working, because `os.proc(scalaCliBinPath, "version")` return `printVersionInfo`. 

`maybeCurrentVersion` can contains only valid `Versions`, which can be parsed by `coursier.core.Version`. 

We use `maybeCurrentVersion` and `newestScalaCliVersion`, to check if current version of scala-cli is out to date - [CommandUtils. isOutOfDateVersion](https://github.com/VirtusLab/scala-cli/blob/main/modules/cli/src/main/scala/scala/cli/commands/CommandUtils.scala#L5)

Reply:
So, how is this creating a problem? I just don’t get it...

Reply:
after  `os.proc(scalaCliBinPath, "version”)` returns `printVersionInfo` perhaps I can add logic for finding the first string matching the version regex as you have in the code.

Reply:
Plus, That code would only get executed in tests, because in normal running circumstance, the scalaCliBinPath is going to be None, so only Constants.version is going to be used.

Reply:
solved it by refactoring and making the version command only print the version.